MyEtherWallet/MyEtherWallet
# No Javascript / Old Browser Blockers

## What This Does

Closes out #79:

* Adds a page blocker for if your browser is missing any of the following features:
  * const / let
  * Local storage
  * Object.assign
  * Flexbox support (Otherwise the site will look broken)
* Adds a page blocker if you have Javascript disabled via a noscript tag

The copy for either of these is totally up or editing, I just threw something together. Please advise @tayvano.

## How to Test

1. Run `npm run build`
2. Use `http-server` or something to serve up the `dist/` folder
3. Use out browserstack credentials (Found in slack) and load the page in an old browser to see the bad browser message
4. Load the page in Chrome, and disable javascript to see the no js message like so:

<img width="379" alt="screen shot 2017-10-08 at 12 10 06 pm" src="https://user-images.githubusercontent.com/649992/31319982-a61fd724-ac21-11e7-8cde-5ac6b829e777.png">


## Janky Implementations

Because a few of the third party modules make it into our vendor bundle with a `const` here and a `let` there, I decided that to keep this as fail-proof as possible, I would do the bad browser check outside of the main code bundle. So you'll find it, along with all of the markup for these, at the bottom of `index.html`. This wasn't ideal, but after having the common or vendor bundles fail for a myriad of reasons, felt like a real gordian knot chop.

Also, because they're not served up via React, I couldn't make a nice common component for these kinds of page blockers. So instead I made a sass mixin, `@mixin cover-message` which applies the styles, but allows for customization and overrides.

Requires #273 to work due to needing to be tested with the `npm run build` version of this.

## Screenshots

<img width="856" alt="screen shot 2017-10-08 at 11 58 51 am" src="https://user-images.githubusercontent.com/649992/31319954-32d5b996-ac21-11e7-8fad-4db134e96bfe.png">

![screen shot 2017-10-08 at 12 01 05 pm](https://user-images.githubusercontent.com/649992/31319956-39371f96-ac21-11e7-9aa1-6a6ea8cfd018.png)
